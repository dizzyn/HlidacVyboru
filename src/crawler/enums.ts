export const COMMITTEES = [
  {
    shorcut: "HV",
    name: "Hospodářský výbor",
    id: 3500,
  },
  {
    shorcut: "KV",
    name: "Kontrolní výbor",
    id: 3600,
  },
  {
    shorcut: "MIV",
    name: "Mandátový a imunitní výbor",
    id: 3700,
  },
  {
    shorcut: "ORGV",
    name: "Organizační výbor",
    id: 3800,
  },
  {
    shorcut: "PV",
    name: "Petiční výbor",
    id: 3900,
  },
  {
    shorcut: "RV",
    name: "Rozpočtový výbor",
    id: 3400,
  },
  {
    shorcut: "UPV",
    name: "Ústavně právní výbor",
    id: 4000,
  },
  {
    shorcut: "VOV",
    name: "Volební výbor",
    id: 4100,
  },
  {
    shorcut: "VB",
    name: "Výbor pro bezpečnost",
    id: 4900,
  },
  {
    shorcut: "VEZ",
    name: "Výbor pro evropské záležitosti",
    id: 500,
  },
  {
    shorcut: "VO",
    name: "Výbor pro obranu",
    id: 5000,
  },
  {
    shorcut: "VSP",
    name: "Výbor pro sociální politiku",
    id: 4300,
  },
  {
    shorcut: "VVVKM",
    name: "Výbor pro vědu, vzdělání, kulturu, mládež a tělovýchovu",
    id: 4500,
  },
  {
    shorcut: "VSR",
    name: "Výbor pro veřejnou správu a regionální rozvoj",
    id: 4400,
  },
  {
    shorcut: "Výboru pro zdravotnictví",
    name: "Výbor pro zdravotnictví",
    id: 3200,
  },
  {
    shorcut: "VZP",
    name: "Výbor pro životní prostředí",
    id: 4600,
  },
  {
    shorcut: "ZAV",
    name: "Zahraniční výbor",
    id: 3300,
  },
  {
    shorcut: "ZEV",
    name: "Zemědělský výbor",
    id: 4700,
  },
] as const;

export type ValuesOf<T extends any[]> = T[number];
export const COMMITTEE_SHORTCUTS = COMMITTEES.map(({ shorcut }) => shorcut);
export const COMMITTEE_NAMES = COMMITTEES.map(({ name }) => name);
export type TCommitteeShortcut = ValuesOf<typeof COMMITTEE_SHORTCUTS>;
export type TCommitteeName = ValuesOf<typeof COMMITTEE_NAMES>;
export type TCommittee = {
  shorcut: string;
  name: string;
  id: number;
};
export const COMMITTEES_PER_NAME = COMMITTEES.reduce(
  (acc, c) => ({ ...acc, [c.name]: c }),
  {} as { [key in TCommitteeName]: TCommittee }
);

export const MONTHS = [
  "ledna",
  "února",
  "března",
  "dubna",
  "května",
  "června",
  "července",
  "srpna",
  "září",
  "října",
  "listopadu",
  "prosince",
];
