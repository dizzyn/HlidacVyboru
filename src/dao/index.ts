import { createHlidacAPIGetLink, createHlidacAPISetLink } from "../utils";
import fetch from "node-fetch";

require("dotenv").config();

export type THlidacData = {
  Id: string;
  datum: string;
  cisloJednani: number;
  vybor: string;
  vyborId: number;
  vyborUrl: string;
  dokumenty: {
    DocumentUrl: string;
    jmeno: string;
    popis: string;
    typ:
      | "Pozvánka"
      | "Zápis z jednání"
      | "Usnesení"
      | "Dokumenty a prezentace"
      | "Zvukový záznam z jednání";
  }[];
  audio: { DocumentUrl: string; jmeno: string }[];
};

const options = {
  headers: {
    Authorization: `Token ${process.env.HLIDAC_API_TOKEN}`,
    "Content-Type": "application/json",
  },
};

export const fetchHlidac = async (id: string): Promise<THlidacData | null> => {
  const res = await fetch(createHlidacAPIGetLink(id), options);

  const json = await res.json();
  if (json.Error) {
    console.log("fetchHlidac error:", createHlidacAPIGetLink(id), json);
    return null;
  }
  console.log("fetchHlidac data:", createHlidacAPIGetLink(id), json);
  return json;
};

export const insertHlidac = async (data: THlidacData) => {
  try {
    const res = await fetch(createHlidacAPISetLink(data.Id), {
      ...options,
      method: "POST",
      body: JSON.stringify(data),
    });

    console.log("res", res);
    const json = await res.json();
    console.log("json", json);
  } catch (e) {
    console.log(e?.error?.Error === "Zaznam nebylo mozno vlozit.");
    console.error(e);
    throw e;
  }
};
