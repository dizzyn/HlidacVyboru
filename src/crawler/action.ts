import crawler from ".";
import documents, { fetchDocumentFromLink, TDocument } from "./documents";
import { COMMITTEE_NAMES, TCommitteeName } from "../enums";
import {
  createHlidacDocLink,
  createHlidacId,
  createURL,
  getDate,
  getNumber,
  getOnlyNodeText,
  removeDate,
  removeNumber,
} from "../utils";
import { fetchHlidac, THlidacData } from "../dao";

export interface THlidacOnlyDoc {
  title: string;
  documentUrl: string;
  hlidacLink: string;
}

export interface THlidacOnlyRecord {
  title: string;
  documentUrl: string;
  hlidacLink: string;
}

const removeDuplicities = (docs: TDocument[]) =>
  docs.reduce(
    (acc: TDocument[], doc) =>
      acc.find(({ documentUrl }) => documentUrl === doc.documentUrl)
        ? acc
        : [...acc, doc],
    []
  );

export interface TActionDetail {
  title: string;
  date: string;
  committee: TCommitteeName;
  number: number;
  documents: TDocument[];
  records: TDocument[];
  hlidacOnlyDocuments: THlidacOnlyDoc[];
  hlidacOnlyRecords: THlidacOnlyRecord[];
  sourceUrl: string;
  hlidacId: string;
  hlidacJson: THlidacData | null;
}

export default (sourceUrl: string) =>
  crawler<TActionDetail>(sourceUrl, async ($) => {
    const title = $("#main-content b").text();
    const desc = ($("#main-content").text() ?? "").trim();
    const date = getDate(title) ?? getDate(desc);
    const documentsHref = $("a:contains('Dokumenty')").attr("href");
    const number = parseInt(
      getNumber(removeDate(title)) ?? getNumber(removeDate(desc)) ?? "",
      10
    );
    const committee = getOnlyNodeText($("#left-column h2")).trim();
    // console.log("a", $("#left-column").html());
    if (!documentsHref) {
      throw `Nepodařilo se získat link na dokumenty (${sourceUrl})`;
    }

    if (!date) {
      throw `Nepodařilo se získat datum (${sourceUrl})`;
    }

    if (!number) {
      throw `Nepodařilo se získat číslo jednání (${sourceUrl})`;
    }

    if (!COMMITTEE_NAMES.includes(committee as any)) {
      throw `Nepodařilo najít název výboru: '${committee}' (${sourceUrl})`;
    }

    const hlidacId = createHlidacId(
      date.trim(),
      number,
      committee as TCommitteeName
    );

    if (!hlidacId) {
      throw `Nepodařilo se vypočítat ID pro hlídač (${sourceUrl})`;
    }

    const hlidacJson = await fetchHlidac(hlidacId);

    const docsRaw = await documents(
      createURL(documentsHref),
      date,
      number,
      hlidacJson
    );

    const docs = docsRaw.filter(({ type }) => type !== "ZAZNAM");
    const records = docsRaw.filter(({ type }) => type === "ZAZNAM");

    const hlidacOnlyDocuments =
      hlidacJson?.dokumenty?.reduce(
        (acc: THlidacOnlyDoc[], doc, i: number) =>
          docs.find(({ documentUrl }) => doc.DocumentUrl === documentUrl)
            ? acc
            : [
                ...acc,
                {
                  title: doc.jmeno,
                  documentUrl: doc.DocumentUrl,
                  hlidacLink: createHlidacDocLink(hlidacId, i),
                },
              ],
        []
      ) ?? [];

    const hlidacOnlyRecords =
      hlidacJson?.audio?.reduce(
        (acc: THlidacOnlyRecord[], rec: any, i: number) =>
          records.find(({ documentUrl }) => rec.DocumentUrl === documentUrl)
            ? acc
            : [
                ...acc,
                {
                  title: rec.jmeno,
                  documentUrl: rec.DocumentUrl,
                  hlidacLink: createHlidacDocLink(hlidacId, i),
                },
              ],
        []
      ) ?? [];

    // Pozvánka přimo z novinky
    await fetchDocumentFromLink(
      docs,
      `a:contains('pozvánka')`,
      "POZVANKA",
      hlidacJson,
      $,
      sourceUrl
    );

    return {
      title: removeDate(removeNumber(title)),
      date: date.trim(),
      committee: committee as TCommitteeName,
      number: number,
      sourceUrl,
      documents: removeDuplicities(docs),
      records: removeDuplicities(records),
      hlidacOnlyDocuments,
      hlidacOnlyRecords,
      hlidacId,
      hlidacJson,
    };
  });
