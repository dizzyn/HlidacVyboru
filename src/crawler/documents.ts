import crawler from ".";
import { THlidacData } from "../dao";

import {
  createDocFileName,
  createHlidacDocLink,
  createURL,
  getDate,
  removeDate,
} from "../utils";

export type TDocumentType =
  | "POZVANKA"
  | "ZAPIS"
  | "ZAZNAM"
  | "USNESENI"
  | "DOKUMENT";

export interface TDocument {
  filename: string;
  desc: string;
  type: TDocumentType;
  documentUrl: string;
  sourceUrl: string;
  hlidacLink: string | null;
}

export const getHlidacDocLink = (
  documentUrl: string,
  hlidacJson: THlidacData | null,
  type: TDocumentType
) => {
  if (!hlidacJson) {
    return null;
  }
  const hlidacDocIndex = hlidacJson[
    type === "ZAZNAM" ? "audio" : "dokumenty"
  ]?.findIndex(({ DocumentUrl }: any) => {
    return DocumentUrl === documentUrl;
  });

  return hlidacDocIndex > -1
    ? type === "ZAZNAM"
      ? "N/A" // no detail of audio at the Hlidac
      : createHlidacDocLink(hlidacJson.Id, hlidacDocIndex)
    : null;
};

export const fetchDocumentFromLink = async (
  documents: TDocument[],
  selector: string,
  type: TDocumentType,
  hlidacJson: THlidacData | null,
  $: cheerio.CheerioAPI,
  sourceUrl: string
) => {
  const $docElements = $(selector);
  for (let i = 0; i < $docElements.length; i++) {
    const $docElement = $($docElements[i]);
    const docHref = $docElement.attr("href");
    if (docHref) {
      const documentUrl = createURL(docHref);
      const desc = (removeDate($docElement.text()) ?? "").trim();
      const definitiveType = desc.toLowerCase().includes("mp3")
        ? "ZAZNAM"
        : type;
      if (docHref.includes("orig2.sqw")) {
        documents.push({
          desc,
          filename: createDocFileName(
            desc,
            definitiveType === "ZAZNAM"
              ? "mp3"
              : docHref.includes("pdf=1")
              ? "pdf"
              : "doc"
          ),
          type: definitiveType,
          documentUrl,
          sourceUrl,
          hlidacLink: getHlidacDocLink(documentUrl, hlidacJson, type),
        });
      } else if (docHref.includes("text2.sqw")) {
        documents.push(
          await loadDocument(createURL(docHref), type, hlidacJson)
        );
      }
    }
  }
};

export const loadDocumentArchive = async (
  documents: TDocument[],
  selector: string,
  date: string,
  type: TDocumentType,
  hlidacJson: THlidacData | null,
  $: cheerio.CheerioAPI,
  sourceUrl: string
) => {
  const href = $(selector).attr("href");

  if (!href) {
    throw new Error(
      `Nepodařilo se získat odkaz na stránku archivu (${sourceUrl})`
    );
  }

  await crawler(createURL(href), async ($) => {
    const nextHref = $(".next").attr("href");

    const $docElements = $(`td:contains(${date.split(" ").join("\u00a0")})`)
      .prev()
      .find("a");

    for (let i = 0; i < $docElements.length; i++) {
      const $docElement = $($docElements[i]);
      const docHref = $docElement.attr("href") ?? "";
      if (docHref) {
        const doc = await loadDocument(
          createURL(docHref),
          type,
          hlidacJson,
          $docElement.parent("td").next("td").text()
        );
        documents.push(doc);
      }
    }

    if (nextHref) {
      await loadDocumentArchive(
        documents,
        ".next",
        date,
        type,
        hlidacJson,
        $,
        sourceUrl
      );
    }
  });
};

export const loadDocument = async (
  sourceUrl: string,
  type: TDocumentType,
  hlidacJson: THlidacData | null,
  desc?: string
) =>
  crawler<TDocument>(sourceUrl, async ($) => {
    const title = $(".page-title h1").text();
    const href =
      $("a:contains('Originál dokumentu')").attr("href") ??
      $("a:contains('Verze PDF')").attr("href");

    if (!title) {
      throw new Error(`Nepodařilo se získat titulek dokumentu (${sourceUrl})`);
    }

    if (!href) {
      throw new Error(`Nepodařilo se získat adresu dokumentu (${sourceUrl})`);
    }

    const documentUrl = createURL("text/" + href);
    return {
      desc: desc ?? removeDate(title),
      filename: createDocFileName(title, "docx"),
      type,
      documentUrl,
      sourceUrl,
      hlidacLink: getHlidacDocLink(documentUrl, hlidacJson, type),
    };
  });

const loadAlternateMetaHref = async (
  sourceUrl: string,
  number: number
): Promise<string | null> =>
  crawler<string | null>(sourceUrl, async ($) => {
    // console.log("Hledám meta", number, sourceUrl);

    const metaHref = $(`a:contains(${number})`).attr("href");
    // console.log("Nalezl jsem", metaHref);
    if (metaHref) {
      return metaHref;
    }

    const nextHref = $(".next").attr("href");
    // console.log("Zkouším next", nextHref);
    if (nextHref) {
      return await loadAlternateMetaHref(createURL(nextHref), number);
    }
    return null;
  });

const loadMeta = async (
  sourceUrl: string,
  number: number,
  hlidacJson: THlidacData | null
) =>
  crawler<TDocument[]>(sourceUrl, async ($) => {
    const documents: TDocument[] = [];

    // Pozvanka z meta
    await fetchDocumentFromLink(
      documents,
      `h4:contains('Pozvánka') + table a:contains(${number})`,
      "POZVANKA",
      hlidacJson,
      $,
      sourceUrl
    );

    // Zápis z meta
    await fetchDocumentFromLink(
      documents,
      `h4:contains('Zápis z jednání') + table a:contains(${number})`,
      "ZAPIS",
      hlidacJson,
      $,
      sourceUrl
    );

    // Dokumenty, prezentace
    await fetchDocumentFromLink(
      documents,
      `h4:contains('Dokumenty a prezentace') + table a`,
      "DOKUMENT",
      hlidacJson,
      $,
      sourceUrl
    );

    // Dokumenty, prezentace
    await fetchDocumentFromLink(
      documents,
      `h4:contains('MP3') + table a`,
      "ZAZNAM",
      hlidacJson,
      $,
      sourceUrl
    );

    // Zaznam
    const $zaznamLink = $(`h4:contains('Zvukový záznam z jednání')+* a`);
    const zaznamHref = $zaznamLink.attr("href");
    if (zaznamHref) {
      const documentUrl = createURL(zaznamHref);

      documents.push({
        desc: `Zvukový záznam z jednání č ${number}.`,
        filename: $zaznamLink.text(),
        type: "ZAZNAM",
        documentUrl,
        sourceUrl,
        hlidacLink: getHlidacDocLink(documentUrl, hlidacJson, "ZAZNAM"),
      });
    }

    return documents;
  });

export default (
  sourceUrl: string,
  date: string,
  number: number,
  hlidacJson: THlidacData | null
) =>
  crawler<TDocument[]>(sourceUrl, async ($) => {
    const metaHref = $(
      `h2:contains('Pozvánky, zvukové záznamy a prezentace ze schůzí')`
    )
      .next()
      .find(`a:contains(${number})`)
      .attr("href");

    const nextMetaHref = $(
      `h2:contains('Pozvánky, zvukové záznamy a prezentace ze schůzí')`
    )
      .next()
      .next()
      .find(`a:contains(další)`)
      .attr("href");

    let documents: TDocument[] = [];

    if (metaHref) {
      documents = [
        ...documents,
        ...(await loadMeta(createURL(metaHref), number, hlidacJson)),
      ];
    } else if (nextMetaHref) {
      const alternateMetaHref = await loadAlternateMetaHref(
        createURL(nextMetaHref),
        number
      );
      // console.log("Alternate meta href", alternateMetaHref);
      if (alternateMetaHref) {
        documents = [
          ...documents,
          ...(await loadMeta(createURL(alternateMetaHref), number, hlidacJson)),
        ];
      }
    }

    // Pozvánka přimo bez Meta
    await fetchDocumentFromLink(
      documents,
      `h2:contains('Pozvánky na schůze') + table a:contains(${number})`,
      "POZVANKA",
      hlidacJson,
      $,
      sourceUrl
    );

    // Usnesení
    const usneseni = (await Promise.all(
      $(`h2:contains('Usnesení')`)
        .next()
        .find("tr")
        .map(async (_, tr) => {
          const $tr = $(tr);
          const trDate = getDate($tr.find("em").text());
          const href = $tr.find("a").attr("href");

          if (!href) {
            throw new Error(
              `Nepodařilo se získat datum usnesení (${sourceUrl})`
            );
          }

          if (!href) {
            throw new Error(
              `Nepodařilo se získat link na usnesení (${sourceUrl})`
            );
          }

          if (trDate === date) {
            return await loadDocument(createURL(href), "USNESENI", hlidacJson);
          }
        })
        .toArray()
    )) as any;

    // Usnesení z archivu
    await loadDocumentArchive(
      documents,
      `h2:contains('Usnesení') + table + p a`,
      date,
      "USNESENI",
      hlidacJson,
      $,
      sourceUrl
    );

    return [...documents, ...usneseni].filter((a) => a);
  });
