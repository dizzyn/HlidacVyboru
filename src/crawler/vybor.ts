import crawler from ".";
import {
  getDate,
  removeDate,
  filterActions,
  getOnlyNodeText,
  mergeActions,
  createURL,
} from "../utils";

export interface TAction {
  title: string;
  date: string;
  href: string;
  desc: string;
}
export interface TVyborDetail {
  title: string;
  actions: TAction[];
  desc: string;
}

export default (sourceUrl: string) =>
  crawler<TVyborDetail>(sourceUrl, ($) => {
    const title = $("h1").text();
    const desc = ($(".news-item.no-date .news-item-title").text() ?? "").trim();

    const actions: TAction[] = $(".news-item.no-date")
      .map((_, item) => {
        const $title = $(item).find(".news-item-title");
        const dateStr =
          getDate($title.text()) ?? getDate(getOnlyNodeText($(item)));
        const $a = $($title).find("a");
        const url = $a.attr("href");
        if (!url) {
          throw new Error(
            `Nepodařilo se získat URL detailu výboru (${sourceUrl})`
          );
        }
        return {
          title: removeDate($title.text()),
          desc: removeDate(getOnlyNodeText($(item)))
            .replace(/(,(\s)?)?viz /, "")
            .trim(),
          href: createURL(url),
          date: dateStr,
        } as TAction;
      })
      .toArray() as any;

    return Promise.resolve({
      actions: mergeActions(filterActions(actions)),
      title,
      desc,
    });
  });
