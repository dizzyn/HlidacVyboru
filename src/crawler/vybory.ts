import crawler from ".";

export interface TVybor {
  title: string;
  href: string;
}

export default (sourceUrl: string) =>
  crawler<TVybor[]>(sourceUrl, ($) => {
    const data = $(".section .link-list > li > a")
      .map((_, a) => {
        return { title: $(a).text(), href: $(a).attr("href") };
      })
      .toArray() as any;

    return data;
  });
