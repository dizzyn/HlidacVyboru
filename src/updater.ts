import { TActionDetail } from "./crawler/action";
import chalk from "chalk";
import fs from "fs";
import { padWithZeroes as padWithZeroes } from "./utils";
import { insertHlidac, THlidacData } from "./dao";
import { COMMITTEES_PER_NAME, HLIDAC_TYPE_PER_TYPE, MONTHS } from "./enums";
import moment from "moment";

const logPath = __dirname + "/../logs";

const logInsert = async (
  runId: string,
  index: number,
  hlidacId: string,
  newData: Partial<THlidacData>,
  dry: boolean
) => {
  const filePath = `${logPath}/${runId}-${padWithZeroes(index, 3)}-INSERT-${
    dry ? "IDLE" : ""
  }-${hlidacId}.json`;

  // console.log("FILE - ", filePath);

  try {
    fs.writeFile(filePath, JSON.stringify(newData), () => {});
  } catch (e) {
    console.error("Could not write insert log file");
  }
};

export const dateToMoment = (date: string) => {
  MONTHS.map((month, i) => {
    date = date.replace(month, `${i + 1}.`);
  });

  return moment(date, "DD. MM. YYYY");
};

export const createHlidacDate = (date: string) => {
  MONTHS.map((month, i) => {
    date = date.replace(month, `${i + 1}.`);
  });

  const m = moment(date, "DD. MM. YYYY");
  return `${m.format("YYYY-MM-DD")}T00:00:00Z`;
};

export const createHlidacData = (data: TActionDetail): THlidacData => {
  return {
    Id: data.hlidacId,
    datum: createHlidacDate(data.date),
    cisloJednani: data.number,
    vec: null,
    vybor: data.committee,
    vyborId: COMMITTEES_PER_NAME[data.committee].id,
    vyborUrl: COMMITTEES_PER_NAME[data.committee].url,
    zapisJednani: null,
    dokumenty: data.documents.map((doc) => ({
      HsProcessType: "documentsave",
      DocumentUrl: doc.documentUrl,
      DocumentPlainText: null,
      jmeno: doc.filename,
      popis: doc.desc ?? "",
      typ: HLIDAC_TYPE_PER_TYPE[doc.type],
    })),
    audio: data.records.map((rec) => ({
      HsProcessType: "documentsave",
      DocumentUrl: rec.documentUrl,
      DocumentPlainText: null,
      jmeno: rec.filename,
      prepisAudia: null,
    })),
  };
};

const dryEngine = {
  insert: async (runId: string, index: number, data: TActionDetail) => {
    console.log(chalk.green(data.hlidacId, " - To be inserted (DRY)"));
    await logInsert(runId, index, data.hlidacId, createHlidacData(data), true);
  },
};

const ENGINE = {
  insert: async (runId: string, index: number, data: TActionDetail) => {
    console.log(chalk.green(data.hlidacId, " - To be inserted"));
    await insertHlidac(createHlidacData(data));
    await logInsert(runId, index, data.hlidacId, createHlidacData(data), false);
  },
};

export const isToFresh = (data: TActionDetail) => {
  const nowIs = moment();
  const thenIs = dateToMoment(data.date).add(1, "day");
  return !thenIs.isBefore(nowIs);
};

export const update = async (
  data: TActionDetail,
  runId: string,
  index: number,
  dry: boolean
) => {
  const engine = dry ? dryEngine : ENGINE;
  if (isToFresh(data)) {
    console.log(chalk.blue(data.hlidacId), " ---- too fresh, let's wait");
    return;
  }
  if (!data.hlidacJson) {
    await engine.insert(runId, index, data);
  } else {
    console.log(chalk.blue(data.hlidacId), " ---- ok");
  }
};
