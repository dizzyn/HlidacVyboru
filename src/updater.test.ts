import { TActionDetail } from "./crawler/action";
import { THlidacData } from "./dao";
import { createHlidacData, createHlidacDate } from "./updater";

const data: TActionDetail = {
  date: "1. srpna 1981",
  committee: "Hospodářský výbor",
  hlidacJson: null,
  title: "",
  number: 12,
  records: [],
  documents: [],
  hlidacOnlyDocuments: [],
  hlidacOnlyRecords: [],
  sourceUrl: "",
  hlidacId: "3200-78-20201209",
};

test("Convert to hlidac data", () => {
  expect(createHlidacData(data)).toMatchSnapshot();
});

test("Convert to hlidac data", () => {
  expect(createHlidacDate("1. ledna 2020")).toBe("2020-01-01-T00:00:00");
  expect(createHlidacDate("13. srpna 1981")).toBe("1981-08-13-T00:00:00");
});
