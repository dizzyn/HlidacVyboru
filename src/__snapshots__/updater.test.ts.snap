// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert to hlidac data 1`] = `
Object {
  "Id": "3200-78-20201209",
  "audio": Array [],
  "cisloJednani": 12,
  "datum": "1981-08-01-T00:00:00",
  "dokumenty": Array [],
  "vybor": "Hospodářský výbor",
  "vyborId": 3500,
  "vyborUrl": "https://www.psp.cz/sqw/hp.sqw?k=3500",
}
`;
